.github/workflows/changelogs/changelogs.py

```diff
@@ -0,0 +1,639 @@
+        """Validate the trackers mentioned in a list of entries
+
+        Checks any possible typos and verifies Bugzilla trackers via the
+        Bugzilla API.
+
+        If a PR number is provided, additionally match the mentioned trackers
+        against the PR's title and comment messages.
+        """
+
+        issues = []
+
+        if self.pr_number and self.git_repo:
+            pr_validation = True
+            pr_trackers = self.get_pr_trackers(self.git_repo, self.pr_number)
+        else:
+            pr_validation = False
+
+        all_trackers = {}
+
+        for entry in entries:
+            # Check for mistyped trackers
+            # Count actual trackers in the entry
+            num_trackers = functools.reduce(lambda x, y: x + len(y), entry.trackers.values(), 0)
+            # Find all tracker-like words
+            if len(re.findall(self.regex.TRACKER_LIKE, entry.entry)) > num_trackers:
+                issues.append(Issue(IssueType.MISTYPED_TRACKER, entry.file, entry.line, entry.end_line, severe=False))
+
+            for kind, trackers in entry.trackers.items():
+                # Collect all trackers in all entries of the changelog
+                if kind not in all_trackers:
+                    all_trackers[kind] = entry.trackers[kind]
+                else:
+                    all_trackers[kind].extend(entry.trackers[kind])
+
+                # Check if all the trackers mentioned in the
+                # changelog entry are also mentioned in the PR
+                if pr_validation:
+                    for t in trackers:
+                        if kind not in pr_trackers or t not in pr_trackers[kind]:
+                            # Tracker not mentioned in the PR
+                            issues.append(Issue(IssueType.WRONG_TRACKER.format(t[0]), entry.file, entry.line, \
+                                                entry.end_line))
+
+            # Check Bugzilla trackers via the API
+            issues.extend(self.validate_bsc(entry))
+
+        # Check if all the trackers mentioned in the
+        # PR are also mentioned in the changelogs
+        if pr_validation:
+            for kind, trackers in pr_trackers.items():
+                for t in trackers:
+                    if kind not in all_trackers or t not in all_trackers[kind]:
+                        issues.append(Issue(IssueType.MISSING_TRACKER.format(t[0])))
+
+        return issues
+
+    def validate(self, file_list: list[str]) -> list[type[Issue]]:
+        """Validates changelogs in the list of files"""
+
+        # Index the list of files by package
+        self.pkg_idx = self.get_pkg_index(file_list)
+        issues = []
+        entries = []
+        for pkg, files in self.pkg_idx.items():
+            # General checks (package/changelog file mismatch)
+            if not files["files"]:
+                # Changelog added but no file is modified
+                issues.append(Issue(IssueType.WRONG_CHLOG, package=pkg))
```

Comment:
Could this happen intentionally? E.g. when we forgot a (second) tracker and want to add it afterwards

Reply:
Sure, but I think it's better fixed with an explicit approval from rel-eng. The test itself should fail anyway.

Reply:
I agree. in general we should not edit old changelogs.  if this happen it should be only  for good reasons.  and if we add additional bsc  we  (releng) need to  also be sure that the related patchinfo will also  include that reference  in a different way  from a standard fixed  bsc. 
@juliogonzalez / @raulillo82 what's your take?

Reply:
Yes, makes sense.